internal/api/v1/application/stage.go

```diff
@@ -396,44 +500,91 @@ func newPipelineRun(app stageParam) *v1beta1.PipelineRun {
 				"app.kubernetes.io/component":  "staging",
 			},
 		},
-		Spec: v1beta1.PipelineRunSpec{
-			ServiceAccountName: "staging-triggers-admin",
-			PipelineRef:        &v1beta1.PipelineRef{Name: "staging-pipeline"},
-			Params:             params,
-			Workspaces: []v1beta1.WorkspaceBinding{
-				{
-					Name:    "cache",
-					SubPath: "cache",
-					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-						ClaimName: app.MakePVCName(),
-						ReadOnly:  false,
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: app.Stage.ID,
+			Labels: map[string]string{
+				"app.kubernetes.io/name":       app.Name,
+				"app.kubernetes.io/part-of":    app.Namespace,
+				"app.kubernetes.io/created-by": app.Username,
+				models.EpinioStageIDLabel:      app.Stage.ID,
+				models.EpinioStageIDPrevious:   app.PreviousStageID,
+				models.EpinioStageBlobUIDLabel: app.BlobUID,
+				"app.kubernetes.io/managed-by": "epinio",
+				"app.kubernetes.io/component":  "staging",
+			},
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32(0),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app.kubernetes.io/name":       app.Name,
+						"app.kubernetes.io/part-of":    app.Namespace,
+						"app.kubernetes.io/created-by": app.Username,
+						models.EpinioStageIDLabel:      app.Stage.ID,
+						models.EpinioStageIDPrevious:   app.PreviousStageID,
+						models.EpinioStageBlobUIDLabel: app.BlobUID,
+						"app.kubernetes.io/managed-by": "epinio",
+						"app.kubernetes.io/component":  "staging",
 					},
-				},
-				{
-					Name:    "source",
-					SubPath: "source",
-					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-						ClaimName: app.MakePVCName(),
-						ReadOnly:  false,
+					Annotations: map[string]string{
+						"config.linkerd.io/skip-outbound-ports": "443",
 					},
 				},
-				{
-					Name: "s3secret",
-					Secret: &corev1.SecretVolumeSource{
-						SecretName: helmchart.S3ConnectionDetailsSecretName,
-						Items: []corev1.KeyToPath{
-							{Key: "config", Path: "config"},
-							{Key: "credentials", Path: "credentials"},
+				Spec: corev1.PodSpec{
+					InitContainers: []corev1.Container{
+						{
+							Name:         "download-s3-blob",
+							Image:        "amazon/aws-cli:2.0.52", //TODO: remove hardcode
```

Comment:
This TODO suggests we want to do this later. But I guess hard coding this is fine, maybe extract to a constant and remove the todo?

Reply:
Will fix.

Reply:
Done. Commit 7eb20b7e67e5b4cff825d27bb2309b3486e1cc68.